tooling/bundler/src/bundle/windows/templates/SemverCompare.nsh

```diff
@@ -0,0 +1,150 @@
+!include "LogicLib.nsh"
+!include "strExplode.nsh"
+
+!macro SemverCompare Ver1 Ver2 return_var
+    Push `${Ver2}`
+    Push `${Ver1}`
+    Call Func_SemverCompare
+    Pop ${return_var}
+!macroend
+
+Function Func_SemverCompare
+    Var /GLOBAL VER1
+    Var /GLOBAL VER2
+    Var /GLOBAL V1
+    Var /GLOBAL V2
+    Var /GLOBAL V3
+    Var /GLOBAL V4
+    Var /GLOBAL V5
+    Var /GLOBAL VR1
+    Var /GLOBAL VR2
+    Var /GLOBAL VR3
+    Var /GLOBAL VR4
+    Var /GLOBAL VR5
+    Var /GLOBAL RET
+
+    StrCpy $VER1 ""
+    StrCpy $VER2 ""
+    StrCpy $V1 ""
+    StrCpy $V2 ""
+    StrCpy $V3 ""
+    StrCpy $V4 ""
+    StrCpy $V5 ""
+    StrCpy $VR1 ""
+    StrCpy $VR2 ""
+    StrCpy $VR3 ""
+    StrCpy $VR4 ""
+    StrCpy $VR5 ""
+    StrCpy $RET ""
+
+    Pop $VER1
+    Pop $VER2
+
+    !insertmacro strExplode $0 '.' $VER1
+    Pop $V1
+    Pop $V2
+    Pop $V3
+    Pop $V5
+    !insertmacro strExplode $0 '-' $V3
+    Pop $V3
+    Pop $V4
+
+    !insertmacro strExplode $0 '.' $VER2
+    Pop $VR1
+    Pop $VR2
+    Pop $VR3
+    Pop $VR5
+    !insertmacro strExplode $0 '-' $VR3
+    Pop $VR3
+    Pop $VR4
+
+    ${If} $V1 > $VR1
+        Goto higher
+    ${EndIf}
+
+    ${If} $V1 < $VR1
+        Goto lower
+    ${EndIf}
+
+    ${If} $V2 > $VR2
+        Goto higher
+    ${EndIf}
+
+    ${If} $V2 < $VR2
+        Goto lower
+    ${EndIf}
+
+    ${If} $V3 > $VR3
+        Goto higher
+    ${EndIf}
+
+    ${If} $V3 < $VR3
+        Goto lower
+    ${EndIf}
+
+    ${If} $V4 == "beta"
```

Comment:
Any chance we can use a generic logic instead of only allowing alpha/beta as prerelease name? So instead of checking for `== "beta"` we check for `!= ""` and check if `$V4` is lower than `$VR4` (with StrCmp) instead of comparing with `"alpha"`.

Reply:
yeah, I wasn't sure how semver treats other preleases other than `beta` and `alpha` so thought I would hard code them. I will try with your suggestions.

Reply:
it's always a string comparison. `alpha < beta < rc` for instance.

Reply:
alright, this is finally fixed. Feel free to merge as is. I will work on the other feature in another PR.